### DISCUSSION TITLE
Getting global config values?

### DISCUSSION THREAD

**Discussion by zmitchell** [GitoxideLabs/gitoxide#1382] [General]
I’m just starting to poke around `gix` so forgive me if this is clearly documented somewhere. How would you read global config values? I see that there’s some config-reading functionality for repos, but what if I want to know a global config value outside of the context of a repo?

For instance, my use case is making a tool to manage a worktrees workflow, and I want to know the default branch name so I can create a directory/worktree named after the default branch.

**Comment by Byron** [comment]
This should do the trick.

```Rust
let repo = gix::open(repo_path)?;
repo.config_snapshot().string_by_key("init.defaultBranch")?;
```

**Comment by zmitchell** [comment]
Even if there’s no repo at `repo_path`? Like I said, I want to know what the global config is even when there’s not a repo yet.

**Reply by Byron** [reply]
Right, then the [`Source`](https://docs.rs/gix-config/latest/gix_config/enum.Source.html) type can be used to get the path to the respective configuration file to open. Then there is [`from_globals()`](https://docs.rs/gix-config/latest/gix_config/struct.File.html#method.from_globals) which is probably more in line with what you need as it incorporates all non-repo files.
